langkit/compile_context.py

```diff
@@ -2124,6 +2124,10 @@ def dependencies(typ):
             # Filter types that are relevant for dependency analysis
             return [t for t in result if t.is_struct_type or t.is_array_type]
 
+        # Force creation of the entity iterator type, so that it is available
+        # later during public APIs generation.
+        CompiledTypeRepo.root_grammar_class.entity.iterator
+
```

Comment:
IIUC, this is needed if and only if there is at least one iterator type for any entity type. So what about forcing this creation in the `IteratorType` constructor when called on a non-root entity type?